src/cpu/test_lockit/field.cc

```diff
@@ -323,6 +326,96 @@ void copyField(const TLColor src[][kHeight], TLColor dst[][kHeight])
     }
 }
 
+TLRensaResult simulate(TLColor field[][kHeight])
+{
+    // Assume all puyos are grounded.
+    // TODO: add DCHECK to check it.
+
+    // parameters necessary to compute score
+    int chain = 0;
+    int long_bonus[TLRensaResult::MAX_RENSA] {};
+    int num_colors[TLRensaResult::MAX_RENSA] {};
+    int num_connected[TLRensaResult::MAX_RENSA] {};   // # of connected puyos
+    int num_connections[TLRensaResult::MAX_RENSA] {};  // # of groups
+    bool quick = false;
+    
+    int bottom[6] {};
+    bool cont = true;
+    while (cont) {
+        cont = false;
+        int point[6][12] {};
+        int rakkaflg[6] {};
+
+        SmallIntSet used_colors;
+        // check connections and vanish
+        for (int i = 0; i < 6; ++i) {
+            for (int j = bottom[i]; j < 12; ++j) {
+                TLColor color = field[i][j];
+                if (color == TLColor::EMPTY)
+                    break;
```

Comment:
Can we break here? Consider this:

```
..BBBB
.RRRRR
```

and (i, j) = (1, 0)
`syou()` removes `R`, so is `B` checked?

If this doesn't happen, could you add this into a test case?